scripts/src/javatests/org/oppia/android/scripts/common/RepositoryFileTest.kt

```diff
@@ -0,0 +1,135 @@
+package org.oppia.android.scripts.common
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.io.File
+
+/** Tests for [RepositoryFile]. */
+class RepositoryFileTest {
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+  }
+
+  @Test
+  fun testRepoFile_fileInAppDirectory_fileShouldBePresentInCollectedFiles() {
+    tempFolder.newFolder("testfiles", "app")
+    val file = tempFolder.newFile("testfiles/app/TestFile")
+
+    val collectedFiles = RepositoryFile.collectSearchFiles("${tempFolder.root}/testfiles/")
+
+    assertThat(collectedFiles).contains(file)
+  }
+
+  @Test
+  fun testRepoFile_fileInDotGitDirectory_fileShouldNotBePresentInCollectedFiles() {
+    tempFolder.newFolder("testfiles", ".git")
+    val file = tempFolder.newFile("testfiles/.git/TestFile")
+
+    val collectedFiles = RepositoryFile.collectSearchFiles("${tempFolder.root}/testfiles/")
+
+    assertThat(collectedFiles).doesNotContain(file)
+  }
+
+  @Test
+  fun testRepoFile_dotKtExpectedExtension_onlyKtFilesShouldBePresentInCollectedFiles() {
+    val xmlFile = tempFolder.newFile("testfiles/TestFile.xml")
+    val kotlinFile = tempFolder.newFile("testfiles/TestFile.kt")
+
+    val collectedFiles = RepositoryFile.collectSearchFiles(
+      repoPath = "${tempFolder.root}/testfiles/",
+      expectedExtension = ".kt"
+    )
+
+    assertThat(collectedFiles).contains(kotlinFile)
+    assertThat(collectedFiles).doesNotContain(xmlFile)
+  }
+
+  @Test
+  fun testRepoFile_specifyExemptionList_filesInExemptionListShouldNotBePresentInCollectedFiles() {
+    tempFolder.newFolder("testfiles", "app")
+    val testFile1 = tempFolder.newFile("testfiles/TestFile1.kt")
+    val testFile2 = tempFolder.newFile("testfiles/app/TestFile2.kt")
+    val testFile3 = tempFolder.newFile("testfiles/app/TestFile3.kt")
+    val testFile4 = tempFolder.newFile("testfiles/TestFile4.kt")
+    val exemptionList = listOf<String>(
+      "TestFile3.kt",
+      "TestFile4.kt"
+    )
+
+    val collectedFiles = RepositoryFile.collectSearchFiles(
+      repoPath = "${tempFolder.root}/testfiles/",
+      exemptionsList = exemptionList
+    )
+
+    assertThat(collectedFiles).contains(testFile1)
+    assertThat(collectedFiles).contains(testFile2)
+    assertThat(collectedFiles).doesNotContain(testFile3)
+    assertThat(collectedFiles).doesNotContain(testFile4)
+  }
+
+  @Test
+  fun testRepoFile_specifyMultipleParameterCombination_collectedFilesShouldComplyWithAllSpecificationsMentioned() { // ktlint-disable max-line-length
```

Comment:
```suggestion
  fun testRepoFile_specifyMultipleParams_collectedFilesComplyWithAllSpecs() {
```

Alternative that fits within the line. There's usually a way to get these method names to fit. :)